src/SprykerMiddleware/Zed/Process/Communication/Plugin/Iterator/AbstractProcessIteratorPlugin.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace SprykerMiddleware\Zed\Process\Communication\Plugin\Iterator;
+
+use Generated\Shared\Transfer\IteratorSettingsTransfer;
+use Iterator;
+use Spryker\Zed\Kernel\Communication\AbstractPlugin;
+use SprykerMiddleware\Zed\Process\Dependency\Plugin\Iterator\ProcessIteratorPluginInterface;
+
+abstract class AbstractProcessIteratorPlugin extends AbstractPlugin implements ProcessIteratorPluginInterface
+{
+    const PLUGIN_NAME = 'SPRYKER_MIDDLEWARE_ABSTRACT_ITERATOR_PLUGIN';
```

Comment:
Why the abstract plugin has the name?

Reply:
Ah, you moved this logic from plugins.
But in this case, it's an easter egg: you can forget to (re)define the constant and will get weird results.
You have two ways: throw an exception if the const is not re-defined OR just move it down to plugins to make it explicit.